File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        Deque<int[]> queue = new ArrayDeque<>();
        for (int i = 0; i < priorities.length; i++) {
            queue.add(new int[] { i, priorities[i] });
        }
        while (!queue.isEmpty()) {
            boolean hasHigher = false;
            int[] cur = queue.pollFirst();

            for (int[] p : queue) {
                if (p[1] > cur[1]) {
                    hasHigher = true;
                    break;
                }
            }

            if (hasHigher) {
                queue.add(cur);
            } else {
                answer++;
                if (cur[0] == location) {
                    return answer;
                }
            }
        }

        return answer;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque
def solution(priorities, location):
    answer = 0
    queue = deque(enumerate(priorities))
    
    while queue:
        now = queue.popleft()
        hasHigher = False
        for q in queue:
            if q[1] > now[1]:
                queue.append(now)
                hasHigher = True
                break
        if not hasHigher:
            answer += 1
            if now[0] == location:
                return answer 
                
    return answer